usaspending_api/search/v2/views/spending_by_award.py

```diff
@@ -335,6 +336,15 @@ def get_agency_database_id(self, code):
             return None
         return agency_id.id
 
+    def get_agency_slug(self, code):
+        if len(str(int(code))) < 3:
+            code = "{zeroes}{code}".format(zeroes=("0" * (3 - len(str(int(code))))), code=int(code))
+        toptier_agency = ToptierAgency.objects.get(toptier_code=code)
+        submission = SubmissionAttributes.objects.filter(toptier_code=code).first()
+        if submission is None or toptier_agency is None:
```

Comment:
Is there a scenario where a `submission` would exist but a `toptier_agency` wouldn't? If not, can `vw_published_dabs_toptier_agency` be used here like was in DEV-5081?